src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -1,15 +1,51 @@
 package ch.fhnw.algd1.converters.binary;
 
 public class BinConverter {
-	public static String toString(int x) {
-		// TODO expect x to be in range [-128, 127], return string with 8 binary
-		// digits representing x in 2-complement
-		return "00000000";
-	}
-
-	public static int fromString(String text) {
-		// TODO expect text to contain 8 binary digits, parse to int value in
-		// 2-complement
-		return 0;
-	}
+public static String toString(int x) {
+
+/*
+    var str = Integer.toBinaryString(x & 0xFF); // it's now unsigned
+
+    var zeros = new String(new char[8 - str.length()]).replace('\0', '0');
+
+    return zeros + str;
+    */
+    
+ 
+    StringBuilder value = new StringBuilder();
+    if (x < 0) x += 256;
+    
+    int i = 1 << 7;
+    while (i >= 1) {
+        if (x >= i) {
```

Comment:
Hier wäre auch noch als Variante denkbar
```suggestion
        if ((x & i) != 0) {
```
was dann den Charm hätte, dass der Wert von `x` gar nicht verändert zu werden braucht (das `x -= i;` also entfallen kann).

Reply:
looks elegant and makes sense, have to try it

Reply:
just parse the value of this to the string and append should work...

Reply:
Ja, ginge, nur ist "in einen String parsen" ja ein Mechanismus in der Bibliothek, der für beliebige Zahlen ausgelegt ist und dementsprechend etwas aufwändiger ist. Hier wäre dann ein einfaches "parsing" auch gerade `value.append((x & i) == 0 ? '0' : '1')`...